src/singleton.jl

```diff
@@ -4,11 +4,11 @@
 A singleton communications context, used for single-process runs.
 [`ClimaComms.CPU`](@ref) and [`ClimaComms.CUDA`](@ref) device options are currently supported.
 """
-struct SingletonCommsContext <: AbstractCommsContext
-    device::AbstractDevice
+struct SingletonCommsContext{T <: AbstractDevice} <: AbstractCommsContext
+    device::T
 end
 
-SingletonCommsContext(device = CPU()) = SingletonCommsContext(device)
+SingletonCommsContext() = SingletonCommsContext(CPU())
```

Comment:
Looks like redefinition is due to conflict with the default constructor.
```
SingletonCommsContext(device::T) where T = SingletonCommsContext{T}(device)
SingletonCommsContext() = SingletonCommsContext(CPU())
```

Reply:
This would work too, but I’m not sure what the first method brings to the table. The first method here doesn’t result in the warning because `T` can be any type, but since I added the type restriction in the struct parameter type, I think we all we need is the second method